### PR TITLE
HLRC: document the index template v2 and component template APIs

### PR DESCRIPTION
**Pull request by andreidan**
Relates to #53101

### CODE REVIEW

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

File: docs/java-rest/high-level/cluster/put_component_template.asciidoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+[id="{upid}-{api}"]
+=== Put Component Template API
+
+The Put Component Template API allows to create or change a component template.
+
+[id="{upid}-{api}-request"]
+==== Put Component Template Request
+
+An +{request}+ specifies the name of the component template and the template definition
```
Comment (dakrone) [inline docs/java-rest/high-level/cluster/put_component_template.asciidoc:15]:
```suggestion
A +{request}+ specifies the name of the component template and the template definition,
```

File: docs/java-rest/high-level/cluster/put_component_template.asciidoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+which can consist of the settings, mappings or aliases, together with a version (which
+can be used to simply component tempalte management by external systems) and a metadata
```
Comment (dakrone) [inline docs/java-rest/high-level/cluster/put_component_template.asciidoc:17]:
```suggestion
can be used to simply component template management by external systems) and a metadata
```

**Review by dakrone — APPROVED** [review]
LGTM, thanks for adding this Andrei! I left two super minor nits

**Comment by andreidan** [comment]
@elasticmachine update branch

**Pull request merged**